packit/cli/update_dist_git.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: MIT
+
+"""
+Update a dist-git repo from a source-git repo
+"""
+
+import pathlib
+import click
+
+from typing import Optional
+
+from packit.config import pass_config
+from packit.config import Config, get_local_package_config
+from packit.api import PackitAPI
+from packit.local_project import LocalProject
+
+
+@click.command("update-dist-git")
+@click.option(
+    "--upstream-ref",
+    type=str,
+    help="Git ref of the last upstream commit in the current branch "
+    "from which packit should generate patches "
+    "(this option implies the repository is source-git).",
+)
+@click.option(
+    "--pkg-tool",
+    type=str,
+    help="""Name or path of the packaging tool used to work with
+    sources in the dist-git repo. A variant of 'rpkg'.
+
+    Currently the implementation of this option is incomplete and
+    'fedpkg' is used to work with sources, regardless of the value given
+    to this option.
+
+    Skip retrieving and uploading source archives to the lookaside cache
+    if not specified.
+    """,
+)
+@click.option(
+    "-m",
+    "--message",
+    metavar="<msg>",
+    type=str,
+    help="""Commit the changes in the dist-git repository and use <msg>
+    as the commit message.
+
+    Mutually exclusive with -F.""",
+)
+@click.option(
+    "-F",
+    "--file",
+    metavar="<file>",
+    type=click.Path(exists=True, dir_okay=False, allow_dash=True),
+    help="""Commit the changes in the dist-git repository and take the commit message
+    from <file>. Use - to read from the standard input.""",
+)
+@click.argument("source-git", type=click.Path(exists=True, file_okay=False))
+@click.argument("dist-git", type=click.Path(exists=True, file_okay=False))
+@pass_config
+def update_dist_git(
+    config: Config,
+    source_git: str,
+    dist_git: str,
+    upstream_ref: Optional[str],
+    pkg_tool: Optional[str],
+    message: Optional[str],
+    file: Optional[str],
+):
+    """Update a dist-git repository using content from a source-git repository
+
+    Update a dist-git repository with patches created from the commits
+    between <upstream_ref> and the current HEAD of the source-git repo.
+
+    This command, by default, performs only local operations and uses
+    the content of the source-git and dist-git repository as it is:
+    does not checkout branches or fetches remotes.
+
+    The source archives are retrieved from the upstream URLs specified in
+    the spec-file and uploaded to the lookaside cache in dist-git only if
+    '--pkg-tool' is specified. Note that the implementation of this option
+    is incomplete so 'fedpkg' is used, regardless of the value specified.
+
+    Example:
+
+    \b
+        $ packit update-dist-git \\
+                -m'Update from source-git' --pkg-tool fedpkg \\
+                src/curl rpms/curl
+    """
+    if message and file:
+        raise click.BadOptionUsage("-m", "Option -m cannot be combined with -F.")
+    if file:
```

Comment:
If I don't specify `-m` nor `-F` (perhaps, I don't want to commit yet), I get:
```python
  File "/home/jpopelka/git/packit/packit/packit/cli/update_dist_git.py", line 109, in update_dist_git
    title, _, message = message.partition("\n\n")
AttributeError: 'NoneType' object has no attribute 'partition'
```

Reply:
Good point. I'm not sure about the behaviour though: should `-m/-F` control whether a commit is created or there should be rather some more explicit flag for this?

My idea was to make these flags very similar to the same flags of `git commit`. In that order of idea, I would say there should be an editor opened if none if them is specified. How does this sound to you?

Reply:
> My idea was to make these flags very similar to the same flags of `git commit`.

Ok, but with `git commit` I already say that I want to commit, while in `packit update-dist-git` the commit sounds like a bonus to me.

> there should be an editor opened if none if them is specified

Nice idea , but I can quit it without leaving any message and we're back to "does that mean no commit at all?".

The help message already says:
```
-m, --message <msg>  Commit the changes and ...
-F, --file <file>    Commit the changes and ...
```
so for me it sounds like "no -m nor -F" -> "no commit".

Reply:
Got the same traceback.

Reply:
> so for me "no (or empty) `-m` nor `-F`" -> "no commit".

Good points. I've tried it out and actually the user experience is quite nice, and I think it will be useful. We can add the "open an editor to ask for a commit message" feature later (or when someone asks for it).

I've updated the help text to be more explicit about this behaviour, and the code to actually fork like this.

Please check it and let me know if this is how you are expecting it to work.